airbyte-migration/src/main/java/io/airbyte/migrate/migrations/MigrationV0_29_0.java

```diff
@@ -0,0 +1,118 @@
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Migrates 2 concepts:
+ * <ul>
+ * <li>Deprecates the default workspace id. It finds all instances of it and sets it to a new
+ * randomly generated id.</li>
+ * <li>Makes namespaceDefinition required in StandardSync. If it is not set, defaults to
+ * destination.</li>
+ * </ul>
+ */
+public class MigrationV0_29_0 extends BaseMigration implements Migration {
+
+  private static final UUID DEFAULT_WORKSPACE_ID = UUID.fromString("5ae6b09b-fdec-41af-aaf7-7d94cfc33ef6");
+  private static final String MIGRATION_VERSION = "0.29.0-alpha";
+
+  private static final ResourceId WORKSPACE_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_WORKSPACE");
+  private static final ResourceId SOURCE_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "SOURCE_CONNECTION");
+  private static final ResourceId DESTINATION_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "DESTINATION_CONNECTION");
+  private static final ResourceId OPERATION_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC_OPERATION");
+  private static final ImmutableSet<ResourceId> RESOURCE_WITH_WORKSPACE_ID = ImmutableSet.of(
+      WORKSPACE_RESOURCE_ID,
+      SOURCE_RESOURCE_ID,
+      DESTINATION_RESOURCE_ID,
+      OPERATION_RESOURCE_ID);
+
+  private static final ResourceId CONNECTION_RESOURCE_ID = ResourceId.fromConstantCase(ResourceType.CONFIG, "STANDARD_SYNC");
+  private static Set<String> NAMESPACE_DEFINITIONS = ImmutableSet.of("source", "destination", "customformat");
+
+  private final Migration previousMigration;
+  private final Supplier<UUID> uuidSupplier;
+
+  public MigrationV0_29_0(Migration previousMigration) {
+    this(previousMigration, UUID::randomUUID);
+  }
+
+  public MigrationV0_29_0(Migration previousMigration, Supplier<UUID> uuidSupplier) {
+    super(previousMigration);
+    this.previousMigration = previousMigration;
+    this.uuidSupplier = uuidSupplier;
+  }
+
+  @Override
+  public String getVersion() {
+    return MIGRATION_VERSION;
+  }
+
+  @Override
+  public Map<ResourceId, JsonNode> getOutputSchema() {
+    return previousMigration.getOutputSchema();
+  }
+
+  @Override
+  public void migrate(Map<ResourceId, Stream<JsonNode>> inputData,
+                      Map<ResourceId, Consumer<JsonNode>> outputData) {
+    final UUID newWorkspaceId = uuidSupplier.get();
+
+    for (final Map.Entry<ResourceId, Stream<JsonNode>> entry : inputData.entrySet()) {
+      final Consumer<JsonNode> recordConsumer = outputData.get(entry.getKey());
+      entry.getValue().forEach(r -> {
+        // if standard sync make sure namespaceDefinition is set. it is now a required property. default to
+        // destination.
+        if (entry.getKey().equals(CONNECTION_RESOURCE_ID)) {
+          if (!r.hasNonNull("namespaceDefinition") || !NAMESPACE_DEFINITIONS.contains(r.get("namespaceDefinition").asText())) {
```

Comment:
Could the original `namespaceDefinition` be something other than those in `NAMESPACE_DEFINITIONS`? If that's possible, the original value should be logged as a warning, in case revert is needed.

Reply:
nope. it shouldn't be possible. i think the enum constraint has always been there. but i'll log it in case my understanding of the history is wrong. it will also help us if we do run into any weird cases, e.g. empty string or what not.